tfjs-backend-webgpu/src/webgpu_util.ts

```diff
@@ -56,6 +58,24 @@ export function computeDispatch(
                    (workGroupSize[2] * elementsPerThread[2])) :
                1
   ];
+
+  if (dispatchX <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE &&
+      dispatchY <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE &&
+      dispatchZ <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE) {
+    return [dispatchX, dispatchY, dispatchZ];
+  }
+
+  util.assert(dispatchX > MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE && !layout.y
+      && !layout.z, () => `Only support flatted layout to reshape dispatch ` +
+      `size when the dispatch size exceeds the limit size on x dimension.`);
+
+  let dispatchAverage = Math.ceil(Math.sqrt(dispatchX));
```

Comment:
This assumes dispatchY and dispatchZ are 1. Is this guaranteed by `!layout.y && !layout.z`?

Reply:
Yes. But I think it is more accurate to check these two variables whether are undefined.